### PR TITLE
Add configuration API for In memory gateway deduplication persistence

### PR DESCRIPTION
**Pull request by ramonsmits**
LRU based to contraint the number of items. Refactored to also remove an O(n) operation.

### CODE REVIEW

File: src/NServiceBus.Core/Persistence/InMemory/Gateway/InMemoryGatewayDeduplicationFeature.cs [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+            //DependsOn<gateway>();
+            Defaults(s => s.EnableFeature(typeof(InMemoryGatewayDeduplicationFeature)));
+        }
+
+        /// <summary>
+        /// See <see cref="Feature.Setup" />.
+        /// </summary>
+        protected internal override void Setup(FeatureConfigurationContext context)
+        {
+            if(!context.Settings.TryGet<int>(MaxSizeKey,out var maxSize))
```
Comment (SzymonPobiega) [inline src/NServiceBus.Core/Persistence/InMemory/Gateway/InMemoryGatewayDeduplicationFeature.cs]: The default max side can be registered as a default value so we don't need to do a check here.

File: src/NServiceBus.Core/Persistence/InMemory/Gateway/InMemoryGatewayDeduplication.cs [hunk 2]
```diff
@@ -2,63 +2,48 @@
...
         }
 
-        public int DeleteDeliveredMessages(DateTime until)
+        [MethodImpl(MethodImplOptions.Synchronized)]
+        public Task<bool> DeduplicateMessage(string clientId, DateTime timeReceived, ContextBag context)
         {
-            int count;
-            lock (persistence)
+            // Return FALSE if item EXISTS, TRUE if ADDED
+            if (set.TryGetValue(clientId, out var existingNode)) // O(1)
```
Comment (SzymonPobiega) [inline src/NServiceBus.Core/Persistence/InMemory/Gateway/InMemoryGatewayDeduplication.cs]: Nice LRU implementation 👍

File: src/NServiceBus.Core/Persistence/InMemory/Gateway/InMemoryGatewayDeduplicationFeature.cs [hunk 3]
```diff
@@ -0,0 +1,32 @@
+﻿namespace NServiceBus.Features
+{
+    using Gateway.Deduplication;
+
+    /// <summary>
+    /// Used to configure in memory outbox persistence.
+    /// </summary>
+    public class InMemoryGatewayDeduplicationFeature : Feature
```
[the remark this answers is not in the document]
Comment (ramonsmits) [reply src/NServiceBus.Core/Persistence/InMemory/Gateway/InMemoryGatewayDeduplicationFeature.cs]: You are right :-) didn't see it :-)

**Comment by ramonsmits** [comment]
Closing in favor for https://github.com/Particular/NServiceBus/pull/5397

**Comment by ramonsmits** [comment]
Awaiting https://github.com/Particular/NServiceBus/pull/5397 to be merged first

**Pull request merged**